### ISSUE TITLE
Deprecated safe_indexing in sklearn 0.22

### ISSUE THREAD

**Issue by troiganto** [skorch-dev/skorch#612]
Indexing a skorch dataset with a list of indices emits a `FutureWarning` originating in sklearn.

Script to reproduce:
```python
import torch, skorch, warnings

# Dummy dataset
print('Skorch', skorch.__version__)
X, y = torch.randn(2, 100, 3)
ds = skorch.dataset.Dataset(X, y)

# Cause of warning
warnings.simplefilter('error')
_ = ds[[0]]
```

Output:
```python
Skorch 0.7.0
Traceback (most recent call last):
  File "repro.py", line 6, in <module>
    _ = ds[[0]]
  File "<prefix>/skorch/dataset.py", line 207, in __getitem__
    Xi = multi_indexing(X, i, self.X_indexing)
  File "<prefix>/skorch/utils.py", line 287, in multi_indexing
    return indexing(data, i)
  File "<prefix>/skorch/utils.py", line 183, in _indexing_other
    return safe_indexing(data, i)
  File "<prefix>/sklearn/utils/deprecation.py", line 87, in wrapped
    warnings.warn(msg, category=FutureWarning)
FutureWarning: Function safe_indexing is deprecated; safe_indexing is deprecated in version 0.22 and will be removed in version 0.24.
```

I've encountered this warning with the Skorch version available on Pip (0.7.0), but from a glance into the Github repo, it looks like this is still in the master branch.

Finally, thank you for maintaining Skorch! I've just recently started using PyTorch and your package reduces the amount of boilerplate I have to write significantly!

**Comment by ottonemo** [comment]
Thanks for reporting :)

This should be fixed in commit ab8e6edc487f0bef8109bfeacaae8e4cb070dac8. 

There is a conditional import in `skorch/utils.py` which will use `sklearn.utils._safe_indexing` instead of `.safe_indexing` for later versions of scikit learn (0.24+).

**Comment by troiganto** [comment]
Alright, thanks for the info! I'm gonna close this issue then, but at least anyone else stumbling over this will (hopefully) find this issue in the future. :+1:
